pkg/proxy/nftables/proxier.go

```diff
@@ -601,6 +606,15 @@ func (proxier *Proxier) setupNFTables(tx *knftables.Transaction) {
 			),
 		})
 	}
+	if proxier.serviceCIDRs.Len() > 0 {
+		tx.Add(&knftables.Rule{
+			Chain: kubeServicesChain,
+			Rule: knftables.Concat(
+				ipX, "daddr", "==", "{", strings.Join(proxier.serviceCIDRs.UnsortedList(), ","), "}",
+				"reject",
```

Comment:
Sorry, I changed my mind about this while we were discussing it on slack, but I guess I wasn't clear :slightly_smiling_face: 

ISTM that traffic to an unused port on an active cluster IP should get rejected, in the same way that (assuming no firewall), traffic to a non-listening port on a host would get rejected.

But traffic to an unused cluster IP should just get dropped, in the same way that traffic to an unused IP on your local network would get dropped/ignored. (Right now we just ignore unassigned clusterIPs, which probably means they end up getting routed along the default route, which means they eventually get dropped by someone, but maybe not as efficiently as they should.)

So there should be two commits:
1. reject connections to inactive ports/protocols on active clusterIPs
2. drop packets to unused clusterIPs